action/copy_test.go

```diff
@@ -65,6 +71,102 @@ func (suite CopyTestSuite) TestCopyObjectToNotSeekableFile() {
 	assert.Equal(suite.T(), size, total)
 }
 
+func (suite CopyTestSuite) TestCopySeekableFileToRemote() {
+	size := int64(1024 * 1024) // 1M
+	f, err := NewMockReadAtCloseSeeker(size)
+	defer f.Close()
+	if err != nil {
+		suite.T().Fatal(err)
+	}
+	total, err := CopySeekableFileToRemote(f, storage.MockMBObject)
+	assert.Nil(suite.T(), err)
+	assert.Equal(suite.T(), size, total)
+}
+
 func TestCopyTestSuite(t *testing.T) {
 	suite.Run(t, new(CopyTestSuite))
 }
+
+// MockReadAtCloseSeeker implements the io.Reader, io.ReaderAt, io.Seeker
+// and io.Closer interfaces by reading from a byte slice.
+// a MockReadAtCloseSeeker is read-only and supports seeking.
+// The zero value for Reader operates like a Reader of an empty slice.
+type MockReadAtCloseSeeker struct {
```

Comment:
We don't need to mock a real `ReadAtCloseSeeker`, `Read` and `ReadAt` could return random data